vivo-project/Vitro
# Upload documents in Vitro|VIVO

**[Github Issue](https://github.com/vivo-project/VIVO/issues/3603)**
Goes with PR in Vitro-languages
# What does this pull request do?
Introduces upload files associated with individuals in Vitro|VIVO 
# What's new?
New File Upload Controller to upload and delete files on file storage
New vitro-public:pulicFilename data property to save uploaded file name.
New vitro-public:storedFile object property. As an example to allow users to attach file to specific individuals. 
List view for this property (can be used for any other custom properties with Range class vitro-public:File).

New config options:
Max allowed file size in bytes
fileUpload.maxFileSize = 30000000 
Allowed upload mime types
fileUpload.allowedMIMETypes = image/png, application/pdf
Adds new dependency on Apache Tika to detect file MIME type.

# How should this be tested?
Set/uncomment config properties (mime tyes and max file size). 
Set domain range for  vitro-public:storedFile property to some individual class, for example http://purl.org/ontology/bibo/Book. Create individual of this class.
Got individual page, find storedFile + icon. Try adding some files to it. After uploading you should see files in list and have option to download files by clicking. Also try removing files by clicking on basket. 

Example:
 This change requires documentation updates: 

- Where to set max file size
- Where to set allowed mime types
- How to create custom object property for file upload and set list view for it.

# Interested parties
@VIVO-project/vivo-committers
